walletsetup.go

```diff
@@ -558,9 +559,30 @@ func createSimulationWallet(cfg *config) error {
 	return nil
 }
 
+// checkCreateDir checks that the path exists and is a directory.
+// If path does not exist, it is created.
+func checkCreateDir(path string) error {
+	if fi, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// Attempt data directory creation
+			if err = os.MkdirAll(path, 0700); err != nil {
+				return fmt.Errorf("cannot create directory: %s", err)
+			}
+		} else {
+			return fmt.Errorf("error checking directory: %s", err)
+		}
+	} else {
+		if !fi.IsDir() {
+			return fmt.Errorf("path '%s' is not a directory", path)
+		}
+	}
+
+	return nil
+}
+
 // openDb opens and returns a *walletdb.DB (boltdb here) given the
```

Comment:
I didn't write this :)

And the comment is correct, it actually does return the interface by pointer (because this is how the db is passed to the wallet config).